File: src/test/java/destiny/core/chinese/ziwei/BigRangeFromMainTest.java
/**
 * Created by smallufo on 2017-04-23.
 */
package destiny.core.chinese.ziwei;

import destiny.core.Gender;
import destiny.core.chinese.YinYangIF;
import org.jooq.lambda.tuple.Tuple;
import org.junit.Test;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import java.util.Locale;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertNotNull;

public class BigRangeFromMainTest {

  private Logger logger = LoggerFactory.getLogger(getClass());

  IBigRange impl = new BigRangeFromMain();
  IHouseSeq seq = new HouseSeqDefaultImpl();

  @Test
  public void testString() {
    assertNotNull(impl.getTitle(Locale.TAIWAN));
    assertNotNull(impl.getTitle(Locale.SIMPLIFIED_CHINESE));
    logger.info("title tw = {} , cn = {}", impl.getTitle(Locale.TAIWAN), impl.getTitle(Locale.SIMPLIFIED_CHINESE));
  }


  @Test
  public void getVageRange() {
    // 陽男順行
    assertEquals(Tuple.tuple(2, 11), impl.getVageRange(House.命宮, 2, YinYangIF.陽, Gender.男, seq));
    assertEquals(Tuple.tuple(12, 21), impl.getVageRange(House.父母, 2, YinYangIF.陽, Gender.男, seq));

    // 陰女順行
    assertEquals(Tuple.tuple(2, 11), impl.getVageRange(House.命宮, 2, YinYangIF.陰, Gender.女, seq));
    assertEquals(Tuple.tuple(12, 21), impl.getVageRange(House.父母, 2, YinYangIF.陰, Gender.女, seq));

    // 陰男逆行
    assertEquals(Tuple.tuple(2, 11), impl.getVageRange(House.命宮, 2, YinYangIF.陰, Gender.男, seq));
    assertEquals(Tuple.tuple(12, 21), impl.getVageRange(House.兄弟, 2, YinYangIF.陰, Gender.男, seq));

    // 陽女逆行
    assertEquals(Tuple.tuple(2, 11), impl.getVageRange(House.命宮, 2, YinYangIF.陽, Gender.女, seq));
    assertEquals(Tuple.tuple(12, 21), impl.getVageRange(House.兄弟, 2, YinYangIF.陽, Gender.女, seq));
  }

}

File: src/test/java/destiny/core/chinese/ziwei/BigRangeFromMainTest.kt
/**
 * Created by smallufo on 2017-04-23.
 */
package destiny.core.chinese.ziwei

import destiny.core.Gender.女
import destiny.core.Gender.男
import destiny.core.chinese.YinYang.陰
import destiny.core.chinese.YinYang.陽
import destiny.core.chinese.ziwei.House.*
import kotlin.test.Test
import kotlin.test.assertEquals

class BigRangeFromMainTest {

  internal var impl: IBigRange = BigRangeFromMain()
  private var seq: IHouseSeq = HouseSeqDefaultImpl()

  @Test
  fun getVageRange() {
    // 陽男順行 , 可參考 https://imgur.com/a/g3D9X
    assertEquals(Pair(2, 11), impl.getVageRange(命宮, 2, 陽, 男, seq))
    assertEquals(Pair(12, 21), impl.getVageRange(父母, 2, 陽, 男, seq))
    assertEquals(Pair(112,121) , impl.getVageRange(兄弟 , 2 , 陽 , 男 , seq))


    // 陰女順行
    assertEquals(Pair(2, 11), impl.getVageRange(命宮, 2, 陰, 女, seq))
    assertEquals(Pair(12, 21), impl.getVageRange(父母, 2, 陰, 女, seq))

    // 陰男逆行
    assertEquals(Pair(2, 11), impl.getVageRange(命宮, 2, 陰, 男, seq))
    assertEquals(Pair(12, 21), impl.getVageRange(兄弟, 2, 陰, 男, seq))

    // 陽女逆行
    assertEquals(Pair(2, 11), impl.getVageRange(命宮, 2, 陽, 女, seq))
    assertEquals(Pair(12, 21), impl.getVageRange(兄弟, 2, 陽, 女, seq))
  }

}
